src/cascadia/TerminalSettingsEditor/SettingContainer.cpp

```diff
@@ -172,12 +138,78 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         }
     }
 
-    hstring SettingContainer::_GenerateOverrideMessageText()
+    // Method Description:
+    // - Updates the override system visibility and text
+    // Arguments:
+    // - <none>
+    void SettingContainer::_UpdateOverrideSystem()
     {
-        if (HasSettingValue())
+        if (const auto& child{ GetTemplateChild(L"ResetButton") })
+        {
+            if (const auto& button{ child.try_as<Controls::Button>() })
+            {
+                if (HasSettingValue())
+                {
+                    // We want to be smart about showing the override system.
+                    // Don't just show it if the user explicitly set the setting.
+                    // If the tooltip is empty, we'll hide the entire override system.
+                    hstring tooltip;
+
+                    const auto& settingSrc{ SettingOverrideSource() };
+                    if (!settingSrc)
+                    {
+                        // no source; we're using the system-default value
+                        tooltip = L"";
+                    }
+                    else if (const auto& profile{ settingSrc.try_as<Model::Profile>() })

```

Comment:
i </3 that SettingContainer knows what a profile is

Reply:
I concur - almost seems like we should have `ProfileSettingContainer : SettingContainer` that allows for reverting to the parent, but only for settings in profiles.

All this is terrible, but let's not pretend our settings model was designed for a GUI.

Reply:
Long-term, we should probably move more of this responsibility over the the view model in a post-#9207 world. But I think we're just not ready for that yet, unfortunately.